x-pack/plugins/index_lifecycle_management/common/types/policies.ts

```diff
@@ -174,6 +174,7 @@ export interface AllocateAction {
 export interface ShrinkAction {
   number_of_shards?: number;
   max_primary_shard_size?: string;
+  allow_write_after_shrink?: boolean;
```

Comment:
Possibly outside the scope of this PR, but it would be nice to leverage ES types here from `@elastic/elasticsearch` rather than maintaining our own manual definition.

Reply:
That's a good call! I looked into the ES types but I don't see a defined one for each ILM phase action as there is a generic type for all actions: `export type IlmActions = any;`
<img width="647" alt="Screenshot 2024-07-16 at 10 18 04" src="https://github.com/user-attachments/assets/f7dbca87-18ec-412e-8aef-cbd7973b6356">

I see there is a `IlmShrinkConfiguration` type but it defines only the `number_of_shards` parameter:
<img width="355" alt="Screenshot 2024-07-16 at 10 20 27" src="https://github.com/user-attachments/assets/ecc8a32f-e4ab-4d79-a599-9dc59a9e9a5d">


I suppose we have taken the route of defining the types in the plugin to make sure all parameters are strongly typed. 🤔

Reply:
Of course, I opened https://github.com/elastic/elasticsearch-specification/issues/2727 for the Es specification repo and https://github.com/elastic/kibana/issues/188462 to track the work for replacing the manual definitions with the ES types. I'll go ahead and merge this PR and we can address these separately.